src/main/java/org/opensearch/knn/index/query/KNNWeight.java

```diff
@@ -256,6 +258,26 @@ private Map<Integer, Float> doANNSearch(
             );
         }
 
+        QuantizationParams quantizationParams = quantizationService.getQuantizationParams(fieldInfo);
+
+        // TODO: Change type of vector once more quantization methods are supported
+        byte[] quantizedVector = null;
+
+        if (quantizationParams != null) {
+            QuantizationConfigKNNCollector tempCollector = new QuantizationConfigKNNCollector();
+            reader.searchNearestVectors(knnQuery.getField(), new float[0], tempCollector, null);
```

Comment:
Should we be adding an instance of check for the reader to be safe?

Reply:
The reader is always a SegmentReader

Reply:
The reader is converted above to segment reader..